api/extension-guides/tools.md

```diff
@@ -0,0 +1,248 @@
+- **Hook up some context/behavior that is contributed by another VS Code extension**. For example, you might have a tool that uses the Git extension to retrieve information about the current repository.
+
+## Tool-calling flow
+
+1. Retrieve the list of relevant tools
+1. Send the request to the LLM, providing the list of tool definitions to consider
+1. The LLM generates a response, which may include one or more requests to invoke a tool
+1. Invoke the tool by using the parameter values provided in the LLM response
+1. Provide the tool responses to the LLM
+1. The LLM generates the final user response, which may incorporate tool responses
+
+### Implement tool calling with the chat extension library
+
+You can use the [`@vscode/chat-extension-utils` library](https://www.npmjs.com/package/@vscode/chat-extension-utils) to simplify the process of calling tools in a chat extension.
+
+Implement tool calling in the `vscode.ChatRequestHandler` function of your [chat participant](/api/extension-guides/chat).
+
+1. Determine the relevant tools for the current chat context. You can access all available tools by using `vscode.lm.tools`.
+
+    The following code snippet shows how to filter the tools to only those that have a specific tag.
+
+    ```ts
+    const tools = request.command === 'all' ?
+        vscode.lm.tools :
+        vscode.lm.tools.filter(tool => tool.tags.includes('chat-tools-sample'));
+    ```
+
+1. Send the request and tool definitions to the LLM by using `sendChatParticipantRequest`.
+
+    ```ts
+    const libResult = chatUtils.sendChatParticipantRequest(
+        request,
+        chatContext,
+        {
+            prompt: 'You are a cat! Answer as a cat.',
+            responseStreamOptions: {
+                stream,
+                references: true,
+                responseText: true
+            },
+            tools
+        },
+        token);
+    ```
+
+    The `ChatHandlerOptions` object has the following properties:
+
+    - `prompt`: (optional) Instructions for the chat participant prompt.
+    - `model`: (optional) The model to use for the request. If not specified, the model from the chat context is used.
+    - `tools`: (optional) The list of tools to consider for the request.
+    - `requestJustification`: (optional) A string that describes why the request is being made.
+    - `responseStreamOptions`: (optional) Enable `sendChatParticipantRequest` to stream the response back to VS Code. Optionally, you can also enable references and/or response text.
+
+1. Return the result from the LLM. This might contain error details or tool-calling metadata.
+
+    ```ts
+    return await libResult.result;
+    ```
+
+The full source code of this [tool-calling sample](https://github.com/microsoft/vscode-extension-samples/blob/main/chat-sample/src/chatUtilsSample.ts) is available in the VS Code Extension Samples repository.
+
+### Implement tool calling with prompt-tsx
```

Comment:
I'd say the other option is to implement tool calling yourself. You could use prompt-tsx for this, but you don't have to